deep_tensor/test/test_tensor.cpp

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) 2025-present WATonomous. All rights reserved.
+//
```

Comment:
TODO: Add test for view to ensure the invalid memory access is handled. either use a shared_ptr to manage the memory or throw an exception stating the original memory was freed

Reply:
\+ other view operations if you want